Bug: TDD RED phase rejects import errors as invalid

## Problem

The RED phase verification treats import errors as invalid, but with import-based TDD scaffolding (#261), import errors ARE the expected RED phase behavior.

```
[N3] Verifying red phase (all tests should fail)...
    Results: 0 passed, 0 failed
    [ERROR] 1 tests had errors (not assertion failures)
```

## Root Cause

`verify_phases.py` lines 167-175:
```python
if error_count > 0:
    print(f"    [ERROR] {error_count} tests had errors (not assertion failures)")
    return {
        "error_message": f"Red phase error: {error_count} tests had syntax/import errors",
        ...
    }
```

## Expected Behavior

Import errors during RED phase should count as valid "failures":
- Module doesn't exist → ImportError → RED (correct!)
- Implementation exists → Import succeeds → tests run → assert False → RED (also correct)
- Implementation complete → Import succeeds → tests run → pass → GREEN

## Fix

Change verify_phases to treat errors as valid RED:
```python
# RED phase: errors (import) or failures (assert) both count as "failing"
total_failures = failed_count + error_count
if total_failures > 0 and passed_count == 0:
    print(f"    Red phase PASSED: {total_failures} tests failed/errored as expected")
    ...
```

## Related

- #261: Changed scaffold to import-based tests
- #257: Blocked by this issue

--- a/agentos/workflows/testing/nodes/verify_phases.py
+++ b/agentos/workflows/testing/nodes/verify_phases.py
@@ -143,7 +143,12 @@ def verify_red_phase(state: TestingWorkflowState) -> dict[str, Any]:
     failed_count = parsed.get("failed", 0)
     error_count = parsed.get("errors", 0)
 
-    # Red phase success = ALL tests fail (none pass)
+    # Issue #263: Import errors are valid RED phase behavior
+    # With import-based TDD scaffolding, ImportError means "module doesn't exist"
+    # which is exactly what RED phase should catch.
+    total_red = failed_count + error_count
+
+    # Red phase success = ALL tests fail or error (none pass)
     if passed_count > 0:
         print(f"    [GUARD] WARNING: {passed_count} tests passed unexpectedly!")
         print("    This may indicate pre-existing implementation.")
@@ -153,7 +158,7 @@ def verify_red_phase(state: TestingWorkflowState) -> dict[str, Any]:
             issue_number=state.get("issue_number", 0),
             workflow_type="testing",
             event="red_phase_unexpected_pass",
-            details={"passed": passed_count, "failed": failed_count},
+            details={"passed": passed_count, "failed": failed_count, "errors": error_count},
         )
 
         return {
@@ -164,17 +169,7 @@ def verify_red_phase(state: TestingWorkflowState) -> dict[str, Any]:
             "next_node": "END",
         }
 
-    if error_count > 0:
-        print(f"    [ERROR] {error_count} tests had errors (not assertion failures)")
-
-        return {
-            "red_phase_output": output,
-            "file_counter": file_num,
-            "error_message": f"Red phase error: {error_count} tests had syntax/import errors",
-            "next_node": "END",
-        }
-
-    if failed_count == 0:
+    if total_red == 0:
         print("    [GUARD] WARNING: No tests ran!")
 
         return {
@@ -184,15 +179,19 @@ def verify_red_phase(state: TestingWorkflowState) -> dict[str, Any]:
             "next_node": "END",
         }
 
-    # Success: all tests failed as expected
-    print(f"    Red phase PASSED: {failed_count} tests failed as expected")
+    # Success: all tests failed or errored as expected
+    # Note: errors include ImportError which is valid TDD RED behavior
+    if error_count > 0:
+        print(f"    Red phase PASSED: {error_count} import errors (module doesn't exist yet)")
+    else:
+        print(f"    Red phase PASSED: {failed_count} tests failed as expected")
 
     log_workflow_execution(
         target_repo=repo_root,
         issue_number=state.get("issue_number", 0),
         workflow_type="testing",
         event="red_phase_complete",
-        details={"failed": failed_count},
+        details={"failed": failed_count, "errors": error_count},
     )
 
     return {

--- a/tests/test_testing_workflow.py
+++ b/tests/test_testing_workflow.py
@@ -1639,8 +1639,12 @@ class TestVerifyPhasesModule:
 
         assert "passed unexpectedly" in result.get("error_message", "")
 
-    def test_verify_red_phase_errors(self, tmp_path):
-        """verify_red_phase blocks when tests have errors."""
+    def test_verify_red_phase_errors_are_valid_red(self, tmp_path):
+        """verify_red_phase accepts import errors as valid RED behavior.
+
+        Issue #263: Import errors mean "module doesn't exist yet" which is
+        exactly what TDD RED phase should catch.
+        """
         from agentos.workflows.testing.nodes.verify_phases import verify_red_phase
 
         audit_dir = tmp_path / "audit"
@@ -1667,7 +1671,9 @@ class TestVerifyPhasesModule:
 
             result = verify_red_phase(state)
 
-        assert "error" in result.get("error_message", "").lower()
+        # Import errors now count as valid RED phase (Issue #263)
+        assert result.get("error_message") == ""
+        assert result.get("next_node") == "N4_implement_code"
 
     def test_verify_red_phase_no_tests_ran(self, tmp_path):
         """verify_red_phase blocks when no tests collected."""

